static/js/filter.js

```diff
@@ -672,6 +672,23 @@ export class Filter {
             }
             return this._sub.name;
         }
+        if (
+            (term_types.length === 2 && _.isEqual(term_types, ["pm-with", "near"])) ||
+            (term_types.length === 1 && _.isEqual(term_types, ["pm-with"]))
+        ) {
+            const emails = this.operands("pm-with")[0].split(",");
+            const names = emails.map((email) => {
+                if (!people.get_by_email(email)) {
+                    return email;
+                }
+                return people.get_by_email(email).full_name;
+            });
+
+            // We use join to handle the addition of a comma and space after every name
+            // and also to ensure that we return a string and not an array so that we
+            // can have the same return type as other cases.
+            return names.join(", ");
+        }
```

Comment:
This can be a follow-up, but it is probably a good idea to make sure this function has a test suite with this `near` example in it; this is the kind of thing that it's really nice to have a test for, since it's a clear input/output story, and it's kinda subtle. (We might just need to add a block to the `filter.js` node test)

Reply:
Oh, you did, just in a later commit in this series! Generally the best practice is to include this sort of test in the commit that makes the functional change; the reason this is really nice is that then a reviewer can read the code change and test change together; that makes it substantially easier to feel good about a change being correct when reading that commit.